Common/config/activeinactivetaskbarappearance.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+#include <string_view>
+#include <optional>
+
+
+#include "rapidjsonhelper.hpp"
+#include "taskbarappearance.hpp"
+
+struct ActiveInactiveTaskbarAppearance : TaskbarAppearance {
+	std::optional<TaskbarAppearance> Inactive;
+
+	ActiveInactiveTaskbarAppearance() = default;
+	ActiveInactiveTaskbarAppearance(std::optional<TaskbarAppearance> inactive, ACCENT_STATE accent, Util::Color color, bool showPeek, bool showLine) :
+		TaskbarAppearance(accent, color, showPeek, showLine),
+		Inactive(std::move(inactive))
+	{ }
+
+	template<typename Writer>
+	inline void Serialize(Writer &writer) const
+	{
+		TaskbarAppearance::Serialize(writer);
+		if (Inactive.has_value()) {
```

Comment:
There's an overload of serialize for optionals, so just `rjh::Serialize(writer, Inactive, INACTIVE_KEY);` works

Reply:
oh nice